MantidQt/SliceViewer/src/ConcretePeaksPresenter.cpp

```diff
@@ -580,7 +580,16 @@ bool ConcretePeaksPresenter::addPeakAt(double plotCoordsPointX,
   alg->initialize();
   alg->setProperty("Workspace", peaksWS);
   alg->setProperty("HKL", std::vector<double>(hkl));
-  alg->execute();
+
+
+// Execute the algorithm
+try {
+    alg->execute();
+} catch (...) {
+    g_log.warning("ConcretePeaksPresenter: Could not add the peak. Make sure "
```

Comment:
Personally, I would have put the bit about it being a "ConcretePeaksPresenter", into a separate debug message, as is not something we expect users observing Warnings to understand, but otherwise looks good.